src/servers/src/auth.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod mysql;
+pub mod postgres;
+pub mod pwd;
+
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+
+use self::mysql::MysqlAuthPlugin;
+use self::postgres::PgAuthPlugin;
+use crate::error::Result;
+
+#[async_trait::async_trait]
+pub trait UserProvider: Send + Sync {
+    fn name(&self) -> String;
+
+    async fn get_user_info(&self, identity: &Identity) -> Result<Option<UserInfo>>;
+}
+
+pub type UserProviderRef = Arc<dyn UserProvider>;
+
+type Username = String;
+type HostOrIp = String;
+
+#[derive(Debug)]
+pub enum Identity {
+    UserId(Username, Option<HostOrIp>),
+}
+
+#[derive(Default, Clone, Serialize, Deserialize)]
```

Comment:
`UserInfo` should manually implement `Default` and return `greptime` as default username